packages/dashboard/src/components/robots/robot-summary.tsx

```diff
@@ -0,0 +1,253 @@
+          (taskState.unix_millis_finish_time - taskState.unix_millis_start_time),
+      1,
+    );
+  }, [taskState]);
+
+  React.useEffect(() => {
+    if (!taskState || !taskState.phases || !taskState.active) {
+      setLocation('Failed to retrieve current location');
+      setDestination('Failed to retrieve robot destination');
+      return;
+    }
+
+    const message = Object.values(taskState.phases)[taskState.active - 1]?.detail;
+
+    if (message) {
+      const regex = /\[place:(.*?)\]/g;
+
+      let match;
+      const waypoints = [];
+
+      // Iterate over all matches found by the regular expression
+      while ((match = regex.exec(message.toString()))) {
+        waypoints.push(match[1]);
+      }
+      setLocation(waypoints[0]);
+      setDestination(waypoints[1]);
+    }
+  }, [taskState]);
+
+  const returnDialogContent = () => {
+    const contents = [
+      {
+        title: 'Assigned Tasks',
+        value: taskState ? taskState.booking.id : 'No task',
+      },
+      {
+        title: 'Est. End Time',
+        value: taskState?.unix_millis_finish_time
+          ? `${new Date(taskState?.unix_millis_finish_time).toLocaleString()}`
+          : '-',
+      },
+    ];
+
+    if (taskState) {
+      contents.push(
+        {
+          title: 'Location',
+          value: location,
+        },
+        {
+          title: 'Destination',
+          value: destination,
+        },
+      );
+    }
+
+    return (
+      <>
+        {contents.map((message, index) => (
+          <div key={index}>
+            <TextField
+              label={message.title}
+              id="standard-size-small"
+              size="small"
+              variant="filled"
+              InputProps={{ readOnly: true, className: classes.textField }}
+              fullWidth={true}
+              multiline
+              maxRows={4}
+              margin="dense"
+              value={message.value}
+            />
+          </div>
+        ))}
+      </>
+    );
+  };
+
+  return (
+    <Dialog
+      PaperProps={{
+        style: {
+          backgroundColor: setTaskDialogColor(robotState?.status),
+          boxShadow: 'none',
+        },
+      }}
+      open={isOpen}
+      onClose={() => {
+        setIsOpen(false);
+        onClose();
+      }}
+      fullWidth
+      maxWidth="sm"
+    >
+      <Grid container mb={1} alignItems="center" spacing={1}>
+        <Grid item xs={10}>
+          <DialogTitle align="center">Robot Summary: {robotState?.name}</DialogTitle>
+        </Grid>
+        <Grid item xs={2}>
+          <Grid container justifyContent="flex-end">
+            <Typography variant="subtitle1">{`${
+              robotState?.battery ? robotState?.battery * 100 : 0
+            }%`}</Typography>
+            <Battery80Icon />
+          </Grid>
+        </Grid>
+      </Grid>
+      <Divider />
+      {taskProgress && (
+        <>
+          <Typography variant="h6" fontWeight="bold" align="center">
```

Comment:
```suggestion
          <Typography variant="body2" fontWeight="bold" ml={3} mt={1}>
```
more for aesthetics, I think centering `Task progress` looks a little odd with the dialog title already centered. I would suggest to reduce its size and shift it to the left, similar to how the other fields' titles are placed


![image](https://user-images.githubusercontent.com/5383623/235607568-6100d483-1450-45bb-81fa-20d1e2cb9a58.png)

Reply:
Thanks for the suggestion!